generics/services/sunbird.js

```diff
@@ -0,0 +1,598 @@
+  * @function
+  * @name callToSunbird
+  * @param requestBody - Logged in user Id.
+  * @param token - Logged in user token.
+  * @param url - url of the api call.
+  * @param requestType - http request method
+  * @returns {JSON} - consist of sunbird service response
+*/
+
+function callToSunbird(requestType,url,token,requestBody ="") {
+
+    return new Promise(async (resolve, reject) => {
+        let options = {
+            "headers": {
+                "content-type": "application/json",
+                "authorization": process.env.AUTHORIZATION,
+                "x-authenticated-user-token": token
+            }
+        };
+
+        if (requestType != "GET") {
+            options['json'] = { request: requestBody };
+        }
+
+        url =process.env.SUNBIRD_BASE_URL + url;
+        if (requestType == "PATCH") {
+            request.patch(url, options, callback);
+        } else if (requestType == "GET") {
+            request.get(url, options, callback);
+        } else {
+            request.post(url, options, callback);
+        }
+
+        function callback(err, data) {
+            if (err) {
+                return reject({
+                    message: CONSTANTS.apiResponses.SUNBIRD_SERVICE_DOWN
+                });
+            } else {
+
+               
+                if(data.statusCode == HTTP_STATUS_CODE.ok.status){
+
+                    if(!data.body.responseCode){
+                        data.body = JSON.parse(data.body);
+                    }
+                    if(data.body && data.body.responseCode  && data.body.responseCode == CONSTANTS.common.OK ){
+                        if(data.body.result){
+                           return resolve(data.body.result);
+                        }
+                    }
+                }else{
+                     data.body = JSON.parse(data.body);
+                    return reject({ message:data.body.params.errmsg });
+                }
+                
+            }
+        }
+
+    });
+}
+
+
+/**
+  * Get learning resources.
+  * @function
+  * @name learningResources
+  * @param token - Logged in user token.
+  * @param limit - page limit for the request 
+  * @param offset - page offset for the request
+  * @param filters - api filters for the request 
+  * @returns {JSON} - consist of learning resources list
+*/
+
+const learningResources = function (token,limit,offset,filters = "") {
+    return new Promise(async (resolve, reject) => {
+
+        try {
+
+        const learningResourceUrl = CONSTANTS.endpoints.GET_RESOURCES;
+        
+        let requestBody = {
+            "source": "web",
+            "name": "Resource",
+            "facets": ["board", "gradeLevel", "subject", "medium"],
+            "filters": {
+                "contentType": ["Resource"],
+            },
+            "limit": limit,
+            "mode": "soft",
+            "offset": offset -1
+          }
+
+          if(filters){
+              if(filters["board"]){
```

Comment:
@rakeshSgr - Would suggest making it dynamic, what the keys are passed in filter object,loop through all and add a filter in request body.